server/router.go

```diff
@@ -0,0 +1,151 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	gonet "net"
+	"net/url"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/net"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/health"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+const getOrchestratorTimeout = 2 * time.Second
+
+var errNoOrchestrators = errors.New("no orchestrators")
+
+type Router struct {
+	uris []*url.URL
+	srv  *grpc.Server
+}
+
+func NewRouter(uris []*url.URL) *Router {
+	return &Router{uris: uris}
+}
+
+func (r *Router) Start(uri *url.URL, serviceURI *url.URL, workDir string) error {
+	listener, err := gonet.Listen("tcp", uri.Host)
+	if err != nil {
+		return err
+	}
+	defer listener.Close()
+
+	certFile, keyFile, err := getCert(serviceURI, workDir)
+	if err != nil {
+		return err
+	}
+
+	creds, err := credentials.NewServerTLSFromFile(certFile, keyFile)
+	if err != nil {
+		return err
+	}
+
+	s := grpc.NewServer(grpc.Creds(creds))
+	r.srv = s
+
+	net.RegisterOrchestratorServer(s, r)
+	grpc_health_v1.RegisterHealthServer(s, health.NewServer())
```

Comment:
This is required for the gRPC server to implement the [gRPC health checking protocol]()https://github.com/grpc/grpc/blob/master/doc/health-checking.md) which allows us to use tools like grpc_health_probe to check the health of the server.